beluga/include/beluga/actions/reweight.hpp

```diff
@@ -68,6 +68,50 @@ struct reweight_base_fn {
     return range;
   }
 
+  /// Overload that tracks likelihood in addition to updating the weights.
+  /**
+   * \tparam ExecutionPolicy An [execution policy](https://en.cppreference.com/w/cpp/algorithm/execution_policy_tag_t).
+   * \tparam Range An [input range](https://en.cppreference.com/w/cpp/ranges/input_range) of particles.
+   * \tparam Model A callable that can compute the importance weight given a particle state.
+   * \tparam likelihoodType The type of the member attribute where the likelihood is stored. This is deduced by the
+   * compiler (usually a double or float).
+   * \tparam particleType The class type of the particle. This is deduced by the compiler from the member pointer.
+   * \param policy The execution policy to use.
+   * \param range An existing range of particles to apply this action to.
+   * \param model A callable instance to compute the weights given the particle states.
+   * \param likelihood_member Pointer to the member attribute of the particle class where the likelihood will be stored.
+   *
+   * For each particle, we multiply the current weight by the new importance weight to accumulate information from
+   * sensor updates. Apart from that, the likelihood is stored in the particle, overwriting the "likelihood_member"
+   * passed.
+   */
+  template <
+      class ExecutionPolicy,
+      class Range,
+      class Model,
+      class LikelihoodType,
+      class ParticleType,
+      std::enable_if_t<std::is_execution_policy_v<std::decay_t<ExecutionPolicy>>, int> = 0>
+  constexpr auto
+  operator()(ExecutionPolicy&& policy, Range& range, Model model, LikelihoodType ParticleType::*likelihood_member) const
```

Comment:
@JesusSilvaUtrera thanks for the PR! Consider the following alternative implementation.

Right above (line 67), you can see the original `reweight` transforming particle weights by multiplying the previous weight with the result of evaluating the model at each particle state. So what about if we introduce a `weight` view that just gets you the result of applying the model? Then, we could have a `reweight` overload that takes those weights instead of the model. Then we could those weights to compute other quantities, like the likelihood of an entire cluster :eyes: 

On a related note, I'd rather not add redundant information to the particle. The `weight` already contains `likelihood` information. But I could be convinced otherwise.